src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -222,9 +251,17 @@ uint32_t CommandRequestImpl::SendHMIRequest(
     request[strings::msg_params] = *msg_params;
   }
 
-  if (!application_manager_.ManageHMICommand(result)) {
-    LOG4CXX_ERROR(logger_, "Unable to send request");
-    SendResponse(false, mobile_apis::Result::OUT_OF_MEMORY);
+  if (use_events) {
+    LOG4CXX_DEBUG(logger_,
+                  "subscribe_on_event " << function_id << " "
+                                        << hmi_correlation_id);
+    subscribe_on_event(function_id, hmi_correlation_id);
+  }
+  if (ProcessHMIInterfacesAvailability(hmi_correlation_id, function_id)) {
+    if (!application_manager_.ManageHMICommand(result)) {
+      LOG4CXX_ERROR(logger_, "Unable to send request");
+      SendResponse(false, mobile_apis::Result::OUT_OF_MEMORY);
+    }
```

Comment:
@dtrunov I reely don't know. It was moved from  https://github.com/smartdevicelink/sdl_core/pull/745/commits/6c54ba3c9434d20751ed3f3706fe29850f328380#diff-8096dfbe93a4cc16e2414dbd6ef6834dL227 
Probably if it fails, it means that we are unable ot alocate no more memory. Any way I don't think I should change this behaviour in scope of this PR